alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/SummaryMessageContentProcessor.java

```diff
@@ -105,60 +109,78 @@ private ProviderMessageContent summarize(final ProviderMessageContent message) {
         return itemsByOperation;
     }
 
-    private LinkedHashSet<ComponentItem> createSummarizedComponentItems(final Set<ComponentItem> componentItemsForOperation) {
-        final LinkedHashSet<ComponentItem> summarizedCategoryItems = new LinkedHashSet<>();
-        final Map<String, ComponentAttributeMap> itemsByShallowKey = collectComponentItemData(componentItemsForOperation);
-        for (final ComponentItem componentItem : componentItemsForOperation) {
-            String summaryKey = createSummaryKey(componentItem.getComponentKeys(), componentItem.getItemsOfSameName());
-            final SortedSet<LinkableItem> summarizedLinkableItems = createSummarizedLinkableItems(itemsByShallowKey.get(summaryKey));
-            try {
-                if (!summarizedLinkableItems.isEmpty()) {
-                    ComponentItem newComponentItem = messageOperationCombiner.createNewComponentItem(componentItem, summarizedLinkableItems);
-                    summarizedCategoryItems.add(newComponentItem);
-                }
-            } catch (AlertException e) {
-                // If this happens, it means there is a bug in the Collector logic.
-                throw new AlertRuntimeException(e);
-            }
+    private LinkedHashSet<ComponentItem> summarizeCategoryComponentCounts(Set<ComponentItem> componentItems) {
+        Map<String, Map<ComponentItemPriority, Set<ComponentItem>>> itemsByCategory = new LinkedHashMap<>();
```

Comment:
Objects like this are very tough to follow. Would we be able to convert this into an object instead?